### PR TITLE
Gh 3138 - Generate Key via appleid

### PR DESCRIPTION
**Pull request by biafra23**
Handles #3130, #3138

Changes proposed in this pull request:
- Add "Sign in with Apple" capability in entitlements file
- Configure Web3Auth verifier per stage 
- Request Apple authorization and generate key
- Continue after successful key generation

### CODE REVIEW

File: Multisig/UI/Safe Management/Create Safe/AppleWeb3AuthLogin.swift [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    }
+}
+
+extension AppleWeb3AuthLogin: ASAuthorizationControllerDelegate {
+
+    func authorizationController(controller: ASAuthorizationController, didCompleteWithAuthorization authorization: ASAuthorization) {
+        switch authorization.credential {
+        case let appleIDCredential as ASAuthorizationAppleIDCredential:
+
+            dump(appleIDCredential, name: "appleIDCredential")
```
Comment (DmitryBespalov) [inline Multisig/UI/Safe Management/Create Safe/AppleWeb3AuthLogin.swift]: what does the dump do?

File: Multisig/UI/Safe Management/Create Safe/AppleWeb3AuthLogin.swift [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+class AppleWeb3AuthLogin: NSObject {
+    var onClose: () -> Void
+
+    init(onClose: @escaping () -> Void) {
+        self.onClose = onClose
+    }
+}
+
+extension AppleWeb3AuthLogin: ASAuthorizationControllerDelegate {
```
Comment (DmitryBespalov) [inline Multisig/UI/Safe Management/Create Safe/AppleWeb3AuthLogin.swift:17]: do I understand correctly that this is not yet ready for the review? I see many debug statements, commented out code, and comments that look like pasted from a demo code

**Comment by biafra23** [comment]
recheck

File: Multisig/UI/Safe Management/Create Safe/AppleWeb3AuthLogin.swift [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
```
Comment (DmitryBespalov) [inline Multisig/UI/Safe Management/Create Safe/AppleWeb3AuthLogin.swift:17]: 👍

File: Multisig/UI/Safe Management/Create Safe/CreateSafeFlow.swift [hunk 4]
```diff
@@ -68,12 +77,29 @@ class CreateSafeFlow: UIFlow {
...
         // TODO: submit create safe tx
     }
 
+    func handleAuthorizationAppleIDButtonPress() {
+        let appleIDProvider = ASAuthorizationAppleIDProvider()
+        let request = appleIDProvider.createRequest()
+        request.requestedScopes = [.fullName, .email]
+
+        appleWeb3AuthLogin = AppleWeb3AuthLogin {
+            LogService.shared.debug("onClose called....")
```
Comment (DmitryBespalov) [inline Multisig/UI/Safe Management/Create Safe/CreateSafeFlow.swift]: debug message left here.. will this code area be worked on further?

**Review by DmitryBespalov — APPROVED** [review]

File: Multisig/UI/Safe Management/Create Safe/CreateSafeFlow.swift (hunk 4 above)
Comment (biafra23) [reply Multisig/UI/Safe Management/Create Safe/CreateSafeFlow.swift]: removed.

File: Multisig/UI/Safe Management/Create Safe/CreateSafeFlow.swift [hunk 5]
```diff
@@ -68,16 +77,38 @@ class CreateSafeFlow: UIFlow {
...
+        let request = appleIDProvider.createRequest()
+        request.requestedScopes = [.fullName, .email]
+
+        appleWeb3AuthLogin = AppleWeb3AuthLogin (
+            authorizationComplete: {
+                let view = SafeCreatingViewController()
+                view.onSuccess = { self.safeCreationSuccess() }
+                self.show(view)
+            }, keyGenerationComplete: { key in
+                print("key: \(key)")
```
Comment (DmitryBespalov) [inline Multisig/UI/Safe Management/Create Safe/CreateSafeFlow.swift]: This line doesn't seem like continuing with the flow on success 🤔

"Continue after successful key generation" for apple login - will it be implemented in this pr or it was not an intention to do it here?

Comment (biafra23) [reply Multisig/UI/Safe Management/Create Safe/CreateSafeFlow.swift]: This should mark the place where safe creation can start. I added a TODO

**Review by DmitryBespalov — APPROVED** [review]

**Comment by biafra23** [comment]
recheck

**Comment by biafra23** [comment]
recheck

**Review by DmitryBespalov — APPROVED** [review]

**Pull request merged**